aten/src/ATen/FunctionalizeFallbackKernel.cpp

```diff
@@ -16,19 +16,23 @@ namespace {
       const auto& ivalue = arguments[idx];
       if (ivalue.isTensor()) {
         auto t = ivalue.toTensor();
-        at::functionalization::impl::sync(t);
-        auto t_new = c10::IValue(at::functionalization::impl::from_functional_tensor(t));
-        (*stack)[arguments_begin + idx] = t_new;
+        if (at::functionalization::impl::isFunctionalTensor(t)) {
+          at::functionalization::impl::sync(t);
+          auto t_new = c10::IValue(at::functionalization::impl::from_functional_tensor(t));
+          (*stack)[arguments_begin + idx] = t_new;
+        }
       } else if (ivalue.isTensorList()) {
         auto tensors = ivalue.toTensorList();
-        at::functionalization::impl::sync(tensors);
-        auto t_new = c10::IValue(at::functionalization::impl::from_functional_tensor(tensors));
-        (*stack)[arguments_begin + idx] = t_new;
+        if (at::functionalization::impl::isFunctionalTensor(tensors)) {
+          at::functionalization::impl::sync(tensors);
+          auto t_new = c10::IValue(at::functionalization::impl::from_functional_tensor(tensors));
+          (*stack)[arguments_begin + idx] = t_new;
+        }
       }
     }
     {
       at::AutoDispatchSkipFunctionalize guard;
```

Comment:
Is the guard needed here?
Outside of torch_dispatch. Is there any case where a call from a backend might trigger functionalization again?

Reply:
yep! (maybe you can get the same behavior by removing the key from the include set here instead, although I just went with adding it to the exclude set).